src/Components/Reports/Util.js

```diff
@@ -0,0 +1,68 @@
+import { get } from 'lodash';
+import { pdfRowKeys, percentageKeys, reportRowKeys, SYSTEMS_REPORT_FILE_NAME } from '../../constants';
+
+export const formatData = (data, type) => {
+
+    const systemsRowsData = [];
+    const rowKeys = type === 'json' ?  reportRowKeys : pdfRowKeys;
+
+    data.map((systemItem) => {
+        let rowData = type === 'json' ? {} : [];
+
+        rowKeys.map((rowKey) =>{
+            let rowValue =  get(systemItem, rowKey, '');
+            rowValue = (rowValue === null || rowValue === -1) ?  'N/A' : rowValue.toString();
+            rowValue = (rowValue !== 'N/A' && percentageKeys.includes(rowKey)) ? `${rowValue}%` : rowValue;
+
+            if (type === 'json') {
+                rowData[rowKey] = rowValue;
+            } else if (type === 'pdf') {
+                rowData.push(rowValue);
+            }
+        });
+
+        systemsRowsData.push(rowData);
+    });
+
+    return systemsRowsData;
+
+};
+
+export const responseToJSONData = (data) => {
+    const systemsRowsData = formatData(data, 'json');
+    return JSON.stringify(systemsRowsData);
+};
+
+export const responseToCSVData = (data) => {
+    const items =  formatData(data, 'json');
+    const replacer = (key, value) => value === null ? 'N/A' : value;
+    const header = Object.keys(items[0]);
+    const csvData = [
+        header.join(','), // header row first
+        ...items.map(row => header.map(fieldName => JSON.stringify(row[fieldName], replacer)).join(','))
```

Comment:
Instead of re-iterating on rows, will it be possible to add third argument to `formatData` function and based on its value function will decide whether it requires to perform `JSON.stringify` per row?
By default you can keep it's value as `false`. 
Example - 
`export const formatData = (data, type, convertToStringfy=false) => { ..}`

Any thoughts on this?

Reply:
Hello @kgaikwad `formatData` basically adds key-value pair from the row data object specified, while this map function is for taking out value of particular key and seperate it using , for CSV data. But in any case we don't need JSON.stringify here as I am using toString in the format data to convert into string value. I have removed the JSON.stringify from here and pushed the changes in the latest commit. Let me know in case of any more suggestions on same.